framework/test/mocha/unit/components/storage/entities/account.js

```diff
@@ -628,6 +630,112 @@ describe('Account', () => {
 		});
 
 		describe('filters', () => {
+			describe('asset_contains', () => {
+				let aliceAccount = null;
+				let bobAccount = null;
+
+				beforeEach(async () => {
+					aliceAccount = new accountFixtures.Account({
+						asset: {
+							name: 'Alice',
+							company: 'Acme Corp',
+						},
+					});
+					bobAccount = new accountFixtures.Account({
+						asset: {
+							name: 'Bob',
+							company: 'Acme Corp',
+							tags: ['blockchain', 'crypto'],
+						},
+					});
+
+					await AccountEntity.create([aliceAccount, bobAccount]);
+				});
+
+				it('should return single result when asset value contains unique property at the top level', async () => {
+					const filters = { asset_contains: { name: 'Alice' } };
+					const accounts = await AccountEntity.get(filters);
+
+					expect(accounts.length).to.be.eql(1);
+					expect(accounts[0].address).to.be.eql(aliceAccount.address);
+				});
+
+				it('should return result when asset value contains given array element', async () => {
+					const filters = { asset_contains: { tags: ['crypto'] } };
+					const accounts = await AccountEntity.get(filters);
+
+					expect(accounts.length).to.be.eql(1);
+					expect(accounts[0].address).to.be.eql(bobAccount.address);
+				});
+
+				it('should return all results when asset value contains multiple entries with a given object at the top level', async () => {
+					const filters = { asset_contains: { company: 'Acme Corp' } };
+					const accounts = await AccountEntity.get(filters, { extended: true });
+					const sortedAccounts = accounts.sort((accountA, accountB) =>
```

Comment:
Instead of manually sorting here, can we not add a `sort` filter in the line above for address?

Reply:
I tried that but storage does not support sorting by json/jsonb fields. I discussed with Shu and we came to the conclusion that sorting can be leaved to the service layer.